### PR TITLE
Add `DistFeatureEliminator` class and `select` module

### PR DESCRIPTION
**Pull request by denver1117**
This PR adds the `DistFeatureEliminator` class along with the instantiation of the `eliminate` module. The `eliminate` module will contain feature selection/elimination techniques that leverage distributed training in spark.

Particularly of interest here is the `DistFeatureEliminator` class which will approximate sklearn's RFECV under certain scenarios. It will likely underperform RFECV, however it can be scaled up with much cleaner parallelization using Spark.

From the doc strings:
```
Similar to ``sklearn.feature_selection.RFECV`` but with distributed
feature set removal and cross validation. Fits the base estimator on 
the input training data once for each left out feature sets according 
to the `step` and the scoring of the features according to an initial 
estimator fit with all features. Uses cross validation to score each 
feature set. Results in the fitted base estimator with the best feature set.

Roughly approximates `RFECV` particularly as `step` approaches 1 and 
`min_features_to_select` approaches 0. The value-add is in the 
embarrassingly parallel nature of the `DistFeatureEliminator` algorithm.
```

Open to comments on the algorithm, implementation, example, unit tests, etc.

This PR will update the `sk-dist` version to `0.1.1`

### CODE REVIEW

File: skdist/distribute/select.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    X_test, y_test = _safe_split(estimator, _drop_col(X, index), y, test, train)
+    estimator_.fit(X_train, y_train, **fit_params)
+    return scorer(estimator_, X_test, y_test)
+
+def _divide_chunks(l, n): 
+    """ Divide iterable into n-sized chunks """
+    for i in range(0, len(l), n):  
+        yield l[i:i + n] 
+
+class DistFeatureEliminator(BaseEstimator, ClassifierMixin):
```
[the remark this answers is not in the document]
Comment (denver1117) [reply skdist/distribute/select.py]: Agreed. Maybe we need a completely new name for the class and/or the module? I wanted to have `Eliminator` in there to represent the corollary to RFE in sklearn. But `DistFeatureSelector` fits better with the `selector` module.

[the remark this answers is not in the document]
Comment (chadfoley36) [reply skdist/distribute/select.py]: Right, I wouldn't mind either the class `DistFeatureSelector` or module `eliminator`

[the remark this answers is not in the document]
Comment (denver1117) [reply skdist/distribute/select.py]: Module name has been changed to `eliminate`

File: skdist/distribute/__init__.py [hunk 2]
```diff
@@ -1 +1 @@
-__all__ = ['search', 'encoder', 'ensemble', 'multiclass', 'tests', 'predict']
+__all__ = ['search', 'encoder', 'ensemble', 'multiclass', 'eliminate', 'tests', 'predict']
```
Comment (chadfoley36) [inline skdist/distribute/__init__.py]: 👌

**Pull request merged**